File: mobile/lib/models/later_list_todo.dart
class LaterListTodo {
  final int id;
  final String text;
  final bool isCompleted;
  final DateTime? completedAt;
  final int position;

  LaterListTodo({
    required this.id,
    required this.text,
    required this.isCompleted,
    this.completedAt,
    required this.position,
  });

  factory LaterListTodo.fromJson(Map<String, dynamic> json) {
    return LaterListTodo(
      id: json['id'] as int,
      text: json['text'] as String,
      isCompleted: json['isCompleted'] as bool,
      completedAt: json['completedAt'] != null
          ? DateTime.parse(json['completedAt'] as String)
          : null,
      position: json['position'] as int,
    );
  }

  Map<String, dynamic> toJson() {
    return {
      'id': id,
      'text': text,
      'isCompleted': isCompleted,
      'completedAt': completedAt?.toIso8601String(),
      'position': position,
    };
  }

  LaterListTodo copyWith({
    int? id,
    String? text,
    bool? isCompleted,
    DateTime? completedAt,
    int? position,
  }) {
    return LaterListTodo(
      id: id ?? this.id,
      text: text ?? this.text,
      isCompleted: isCompleted ?? this.isCompleted,
      completedAt: completedAt ?? this.completedAt,
      position: position ?? this.position,
    );
  }
}

File: backend/todue/src/main/java/com/ethan/todue/model/LaterListTodo.java
package com.ethan.todue.model;

import jakarta.persistence.*;
import jakarta.validation.constraints.NotNull;
import lombok.Data;
import lombok.NoArgsConstructor;
import lombok.AllArgsConstructor;
import org.hibernate.annotations.CreationTimestamp;
import org.hibernate.annotations.UpdateTimestamp;

import java.time.Instant;

@Entity
@Table(name = "later_list_todos",
    indexes = {
        @Index(name = "idx_later_list_todos_list_id", columnList = "list_id")
    }
)
@Data
@NoArgsConstructor
@AllArgsConstructor
public class LaterListTodo {

    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private Long id;

    @NotNull
    @ManyToOne(fetch = FetchType.LAZY)
    @JoinColumn(name = "list_id", nullable = false, foreignKey = @ForeignKey(name = "fk_later_list_todo_list",
        foreignKeyDefinition = "FOREIGN KEY (list_id) REFERENCES later_lists(id) ON DELETE CASCADE"))
    private LaterList list;

    @NotNull
    @Column(nullable = false, length = 500)
    private String text;

    @Column(nullable = false, columnDefinition = "BIT(1) DEFAULT 0")
    private Boolean isCompleted = false;

    @Column
    private Instant completedAt;

    @Column(nullable = false)
    private Integer position = 0;

    @Version
    @Column(nullable = false)
    private Long version = 0L;

    @CreationTimestamp
    @Column(nullable = false, updatable = false, columnDefinition = "DATETIME(6) DEFAULT CURRENT_TIMESTAMP(6)")
    private Instant createdAt;

    @UpdateTimestamp
    @Column(nullable = false, columnDefinition = "DATETIME(6) DEFAULT CURRENT_TIMESTAMP(6) ON UPDATE CURRENT_TIMESTAMP(6)")
    private Instant updatedAt;
}
